test/test-parse-expression.cpp

```diff
@@ -1711,6 +1711,43 @@ TEST_F(Test_Parse_Expression, optional_tagged_template_literal) {
   }
 }
 
+TEST_F(Test_Parse_Expression, Diag_Expected_Expression_In_Template_Literal) {
+  {
+    Test_Parser p(u8"`${}`"_sv, capture_diags);
+    p.parse_and_visit_expression();
+    assert_diagnostics(
+        p.code, p.errors,
+        {
+            u8"Diag_Expected_Expression_In_Template_Literal"_diag,
```

Comment:
Must fix: Include location information in the assertion (leading `^^^` or \`). I think your implementation currently puts the diagnostic in a weird spot.